File: binV6/second_problem.rb
require 'json'

posts = {
  "post_1": { "tags": [ "ruby", [ "cli", "json" ], "parsing" ] },
  "post_2": { "tags": [ "ruby", [ "api", [ "cli" ] ] ] }
}

tags = posts.values

collated_tags = tags.flat_map do |data|
  data[:tags].flatten
end.uniq

print JSON.pretty_generate({ data: collated_tags })

File: binV8/second_problem.js
const posts = {
  post_1: { tags: ["ruby", ["cli", "json"], "parsing"] },
  post_2: { tags: ["ruby", ["api", ["cli"]]] },
};

const tags = Object.values(posts).flatMap((post) => {
  return post.tags.flat(Infinity);
});

const filteredTags = [...new Set(tags)];

console.log(JSON.stringify({ data: filteredTags }, null, 2));
